File: ios/Sources/VpnDetectorPluginPlugin/VpnDetectorPluginPlugin.swift
import Foundation
import Capacitor

/**
 * VPN Detector Plugin for Capacitor
 * Detects VPN connections on iOS devices
 */
@objc(VpnDetectorPluginPlugin)
public class VpnDetectorPluginPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "VpnDetectorPluginPlugin"
    public let jsName = "VpnDetectorPlugin"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "isVpnActive", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = VpnDetectorPlugin()

    @objc func isVpnActive(_ call: CAPPluginCall) {
        let isActive = implementation.isVpnActive()
        call.resolve([
            "value": isActive
        ])
    }
}

File: android/src/main/java/com/vpn/detector/plugin/VpnDetectorPluginPlugin.java
package com.vpn.detector.plugin;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;
import com.getcapacitor.annotation.Permission;
import android.Manifest;

@CapacitorPlugin(
    name = "VpnDetectorPlugin",
    permissions = {
        @Permission(strings = { Manifest.permission.ACCESS_NETWORK_STATE }, alias = "network")
    }
)
public class VpnDetectorPluginPlugin extends Plugin {

    private VpnDetectorPlugin implementation;

    @Override
    public void load() {
        implementation = new VpnDetectorPlugin(getContext());
    }

    @PluginMethod
    public void isVpnActive(PluginCall call) {
        try {
            boolean isActive = implementation.isVpnActive();
            
            JSObject ret = new JSObject();
            ret.put("value", isActive);
            call.resolve(ret);
        } catch (Exception e) {
            call.reject("Error detecting VPN status", e);
        }
    }
}
